pkg/controllers/subnet/subnet_poll.go

```diff
@@ -0,0 +1,229 @@
+		return
+	}
+
+	// Group subnets by associatedResource
+	for _, subnet := range subnetList.Items {
+		associatedResource := subnet.Annotations[common.AnnotationAssociatedResource]
+		namespacedName := types.NamespacedName{
+			Namespace: subnet.Namespace,
+			Name:      subnet.Name,
+		}
+		resourceMap[associatedResource] = append(resourceMap[associatedResource], namespacedName)
+	}
+
+	// Process each unique associatedResource
+	for associatedResource, namespacedNames := range resourceMap {
+		ctx := context.Background()
+		log.Info("Polling shared Subnets", "AssociatedResource", associatedResource, "SubnetCount", len(namespacedNames))
+
+		// Update the nsxSubnetCache with the latest NSX subnet data and status list
+		// This is done here to ensure the cache is updated during polling
+		var nsxSubnet *model.VpcSubnet
+		var statusList []model.VpcSubnetStatus
+		var err error
+
+		// Get NSX subnet from API (not from cache during polling to ensure fresh data)
+		log.Info("Fetching NSX subnet during polling", "AssociatedResource", associatedResource)
+		nsxSubnet, err = r.SubnetService.GetNSXSubnetByAssociatedResource(associatedResource)
+		if err != nil {
+			r.handleNSXSubnetError(ctx, err, namespacedNames, associatedResource, "NSX subnet")
+			continue
+		}
+
+		// Get subnet status from NSX (not from cache during polling to ensure fresh data)
+		log.Info("Fetching status list during polling", "AssociatedResource", associatedResource)
+		statusList, err = r.SubnetService.GetSubnetStatus(nsxSubnet)
+		if err != nil {
+			r.handleNSXSubnetError(ctx, err, namespacedNames, associatedResource, "subnet status")
+			continue
+		}
+
+		// Update the cache with the latest NSX subnet and status list
+		r.SubnetService.UpdateNSXSubnetCache(associatedResource, nsxSubnet, statusList)
+
+		// Enqueue all subnet CRs associated with this resource for reconciliation
+		for _, namespacedName := range namespacedNames {
+			log.Info("Enqueueing shared Subnet for reconciliation", "Subnet", namespacedName, "AssociatedResource", associatedResource)
+			r.enqueueSubnetForReconciliation(ctx, namespacedName)
+		}
+	}
+
+	for associatedResource := range r.SubnetService.NSXSubnetCache {
+		if _, ok := resourceMap[associatedResource]; !ok {
+			log.Info("Remove Subnet from cache", "AssociatedResource", associatedResource)
+			r.SubnetService.RemoveSubnetFromCache(associatedResource, "no valid subnets")
+		}
+	}
+}
+
+// enqueueSubnetForReconciliation enqueues a subnet CR for reconciliation by the Subnet Controller
+func (r *SubnetReconciler) enqueueSubnetForReconciliation(ctx context.Context, namespacedName types.NamespacedName) {
+	// Get the subnet CR
+	subnetCR := &v1alpha1.Subnet{}
+	if err := r.Client.Get(ctx, namespacedName, subnetCR); err != nil {
+		log.Error(err, "Failed to get Subnet CR during polling", "Subnet", namespacedName)
+		return
+	}
+
+	// Skip if the subnet is being deleted
+	if !subnetCR.DeletionTimestamp.IsZero() {
+		return
+	}
+
```

Comment:
Maybe skip these check as the reconcile will also check those?

Reply:
It’s better to check so that reconciliation doesn’t have to be triggered all the time, saving some resources.